src/flownet/ahm/_ahm_iteration_analytics.py

```diff
@@ -271,35 +272,43 @@ def make_dataframe_simulation_data(path, eclbase_file, keys):
     # Prepare dataframe
     # pylint: disable-msg=too-many-locals
     df_sim = pd.DataFrame()
-    for id_real, runpath in enumerate(realizations_dict.keys()):
+    id_real = 0
+    for runpath in realizations_dict:
         df_tmp = pd.DataFrame()
-        dates = realizations_dict[runpath].dates
-        if id_real == 0:
-            df_sim["DATE"] = pd.Series(dates)
-            df_sim["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
-        df_tmp["DATE"] = pd.Series(dates)
-        df_tmp["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
-
-        if id_real == 0:
-            for counter, k in enumerate(keys):
-                if counter == 0:
-                    key_list_data = [
-                        x for x in realizations_dict[runpath] if x.startswith(k)
-                    ]
-                else:
-                    key_list_data.extend(
-                        [x for x in realizations_dict[runpath] if x.startswith(k)]
-                    )
+        if (realizations_dict[runpath] is not None) and (
+            np.datetime64(realizations_dict[runpath].dates[-1]) >= end_date
+        ):
+            dates = realizations_dict[runpath].dates
+            if id_real == 0:
+                df_sim["DATE"] = pd.Series(dates)
+                df_sim["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
+            df_tmp["DATE"] = pd.Series(dates)
+            df_tmp["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
```

Comment:
```suggestion
            df_sim["DATE"] = pd.Series(realizations_dict[runpath].dates)
            df_sim["REAL_ID"] = id_real 
```

if df_tmp is need you could do

`df_tmp = df_sim.copy(deep=True)`

Reply:
`df_tmp` is not just a copy, it's a temporary auxiliary dataframe to construct `df_sim`, which is a stacked dataframe